### DISCUSSION TITLE
Bagaimana Cara Sort "Order.Date"?

### DISCUSSION THREAD

**Discussion by BIsmujati** [teamalgoritma/community#532] [Q&A] [answered]
Halo, bolehkah minta tolong, saya sedang mengerjakan LBB pada EDA_Retail pada link https://rpubs.com/wkania/EDA_Retail dengan menggunakan data retail.csv. Sampai pada pertanyaan "3. Berapa kali pembeli mendapatkan diskon 80% dan kapan itu terjadi?". Sebenarnya sudah bisa terlihat pada cara seperti terlihat di gambar ini:

![image](https://user-images.githubusercontent.com/119525681/216805292-5b3cd226-1404-4ccf-a4c3-a31625675989.png)

Tapi, saya ingin lebih lanjut lagi untuk **sort** "Order.Date" sehingga dapat lebih mudah melihatnya. Saya sudah coba cara sebagai berikut:

![image](https://user-images.githubusercontent.com/119525681/216805419-ce8520d6-e1a6-4c69-8ff8-c48c98f2304e.png)

tapi tangga-tanggal yang keluar tidak rapih dalam bentuk tabel yang dapat dibaca dengan mudah.

Pertanyaan:

1. Adakah cara untuk dapat men-**sort** "Order.Date" dalam bentuk tabel yang rapih sehingga lebih mudah terbaca?
2. Kemudian, adakah cara agar "Order.Date" yang ditampilkan tersebut ter-rekap secara rapih per tanggalnya berapa kali sudah per tanggal order yang mendapatkan diskon 80%?

Terima kasih atas jawabannya.

**Comment by onlyphantom** [comment, marked as answer]
Hey Pak Bagus, 

You're doing really well up to this point. Here's what you did:

```
dis <- Elipse[Elipse$Discount == 0.8, c("Order.Date", "Discount")]
# inspecting the first 6 rows shows that the df is not ordered yet
head(dis)
# output:
    Order.Date Discount
15  2015-11-22      0.8
16  2015-11-22      0.8
76  2017-12-09      0.8
102 2016-08-29      0.8
170 2014-09-08      0.8
175 2014-09-14      0.8
```

To order `dis` by `Order.Date`, use `order`. Don't forget to reassign it back to `dis`:
```
dis <- dis[order(dis$Order.Date), ]
head(dis)
# output:
     Order.Date Discount
742  2014-01-04      0.8
7662 2014-01-07      0.8
8981 2014-02-06      0.8
1333 2014-02-16      0.8
1334 2014-02-16      0.8
2883 2014-02-21      0.8
```

Of course, to answer how many observations of transactions with 80% discount, you can just return the number of rows in `dis`: `nrow(dis)`.

If you'd like to see, instead, how many 80% discount happens by each day, you can use the `table` function and then turn it into a data frame:
```
bydates <- data.frame(table(dis$Order.Date))
tail(bydates)
# output:
> tail(bydates)
          Var1 Freq
224 2017-12-11    1
225 2017-12-17    1
226 2017-12-22    1
227 2017-12-23    2
228 2017-12-25    3
229 2017-12-28    1
```

You can also order this `bydates` using `order` just like how it was done above.

Hope this helps! Have a good weekend!

**Reply by BIsmujati** [reply]
Hi, Pak

Thanks for the quick response. Really appreciate it. 
Oh yeah, perhaps I wasn't really clear with my number 2 question. 
I meant was, 
With the already ordered "Order.Date", how can I group similar dates and show how many times the 80% discount appear in each date?

Thanks in advance

**Reply by onlyphantom** [reply]
Hey Pak @BIsmujati  I believe the second part of my answer addresses that. 

Did you see the `bydates` dataframe? That's basically grouping by date. Each row is a date and the number of times 80% discount happen. Does that help? :)

**Reply by BIsmujati** [reply]
Oops, sorry, missed that. Thanks Pak!
